paasta_tools/setup_kubernetes_job.py

```diff
@@ -223,6 +230,31 @@ def setup_kube_deployments(
         else (_, None)
         for _, service_instance in service_instance_configs_list
     ]
+
+    def sort_key(ok_app: Tuple[bool, Optional[Application]]) -> int:
+        """This will return 1 if the desired git_sha matches an existing git_sha, and 0 otherwise. This will cause applications that need a new git_sha
+        to be handled first.
+        This prioritizes "real" bounces (developers pushing new service versions) over things that are likely to be big bounces (config-only changes).
+        Most of the time, this won't matter much, as we should get through our backlog quickly, but when there is a backlog, we want to avoid blocking developers.
+        """
+        ok, app = ok_app
+        if ok:
+            if app.kube_deployment.git_sha in existing_git_shas.get(
+                (
+                    app.kube_deployment.service,
+                    app.kube_deployment.instance,
+                    app.kube_deployment.namespace,
+                ),
+                [],
+            ):
+                return 1
+            else:
+                return 0
+        else:
+            return 2
```

Comment:
just to confirm that my brain isn't too broken: this is to have us handle "broken" apps after everything else so that we don't use any of our rate-limit on 'em?

Reply:
yep, exactly

Reply:
Though actually, the body of the for loop is wrapped in `if app:`, so broken apps don't count against the rate limit anyway. This is mostly here so attribute accesses on `app` don't fail.

I guess I should refactor this to use `if app:` instead of `if ok:`, to mirror the main for loop.

Reply:
I added comments to help make this more readable.